### PR TITLE
refactor repositories

### PR DESCRIPTION
**Pull request by JoGir**
depends on https://github.com/Qabel/qabel-core/pull/530

### CODE REVIEW

**Comment by julianseeger** [comment]
Didn't anyone already.move the Repos to the core?

**Comment by JoGir** [comment]
yes, this pr is about replacing the repositories in android with core repositories

**Comment by julianseeger** [comment]
please rebase/merge

File: qabelbox/src/test/java/de/qabel/qabelbox/chat/interactor/ChatUseCaseTest.kt [hunk 1]
```diff
@@ -11,69 +13,79 @@ import de.qabel.qabelbox.chat.dto.ChatMessage
...
+        transformer = ChatMessageTransformer(identityRepo, contactRepo)
+        chatService = spy(MainChatService(dropConnector, identityRepo, contactRepo, chatDropRepository, InMemoryDropStateRepository()))
     }
 
     @Test
     fun testNoMessages() {
-        val useCase = TransformingChatUseCase(identity, contact, transformer, chatService, mock())
+        val useCase = TransformingChatUseCase(identity, contact, transformer, chatService, chatDropRepository)
         var list: List<ChatMessage>? = null
         useCase.retrieve().toList().toBlocking().subscribe({
```
Comment (audax) [inline qabelbox/src/test/java/de/qabel/qabelbox/chat/interactor/ChatUseCaseTest.kt]: useCase.retrieve().toList().toBlocking().first()

**Comment by JoGir** [comment]
WIP -> core PR

**Comment by JoGir** [comment]
fixes #272, #512 , #600, #514

**Pull request merged**